### ISSUE TITLE
Assignment of fonts after installation of a font with Heading (All) incorrect

### ISSUE THREAD

**Issue by hagege** [WordPress/gutenberg#60245]
### Description

I have uploaded fonts and use the font type (Font Handle in the example) for the headings. However, I have noticed that H2 is displayed correctly with the font, but H3 is not, although I have specified "All". This does not seem to be correct. 

### Step-by-step reproduction instructions

1. Apperance
2. Editor
3. Styles
4. Typography
5. Fonts
6. Choose uploaded Font (in this example Handlee)
7. Headings, choose "All"
8. H1 an H2 are ok, but H3 has the font "Instrument Sans"

### Screenshots, screen recording, code snippet

![issue_font_library_header](https://github.com/WordPress/gutenberg/assets/27292550/19357d58-acf8-4e2d-b50e-562170baddc1)

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by creativecoder** [comment]
If I'm following the screen capture correctly, what I'm seeing is that the H3 block has a block level setting that uses the "Instrument Sans" font.

A block level setting will override global styles settings. If you remove the block font family setting, it should then use the global styles setting for the "Handlee" font.

**Comment by hagege** [comment]
>  If I'm following the screen capture correctly, what I'm seeing is that the H3 block has a block level setting that uses the "Instrument Sans" font.

That is correct. But this is a fresh WordPress instance with TT4. Then I suspect that this setting is related to TT4. I think, at least in this case (and probably also with other block themes) this may be an unexpected behavior for the user.
